src/foam/approval/ApprovableApprovalRequestsRule.js

```diff
@@ -0,0 +1,45 @@
+foam.CLASS({
+  package: 'foam.approval',
+  name: 'ApprovableApprovalRequestsRule',
+
+  documentation: `
+    A rule to update the approvable once it's related approval request has been
+    APPROVED or REJECTED
+  `,
+
+  javaImports: [
+    'foam.core.ContextAwareAgent',
+    'foam.core.X',
+    'foam.dao.DAO',
+    'foam.core.FObject',
+    'foam.nanos.ruler.Operations',
+    'foam.approval.ApprovalRequest',
+    'foam.approval.ApprovalStatus',
+    'foam.approval.Approvable'
+  ],
+
+  implements: ['foam.nanos.ruler.RuleAction'],
+
+  methods: [
+    {
+      name: 'applyAction',
+      javaCode: `
+        ApprovalRequest request = (ApprovalRequest) obj.fclone();
+
+        agency.submit(x, new ContextAwareAgent() {
+          
+          @Override
+          public void execute(X x) {
+            DAO approvableDAO = (DAO) getX().get("approvableDAO");
```

Comment:
where is the update to the FOAM services file? You'll need to add the 'approvableDAO' to the services file so it is available.

Reply:
Done.